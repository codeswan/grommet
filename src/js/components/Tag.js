// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React, { Component } from 'react';
import Anchor from './Anchor';
import Props from '../utils/Props';
import CSSClassnames from '../utils/CSSClassnames';

const CLASS_ROOT = CSSClassnames.TAG;

export default class Tag extends Component {

  render () {
    let classes = [CLASS_ROOT];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    var other = Props.pick(this.props, Object.keys(Anchor.propTypes));

    return (
      <div className={classes.join(' ')} onClick={this.props.onClick}>
        {this.props.children}
        <Anchor {...other} className={`${CLASS_ROOT}--label`}>
          <span>{this.props.label}</span>
        </Anchor>
      </div>
    );
  }
}

Tag.propTypes = {
  label: React.PropTypes.string,
  ...Anchor.propTypes
};
