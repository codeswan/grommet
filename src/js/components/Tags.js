// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React, { Component } from 'react';
import Box from './Box';
import Props from '../utils/Props';
import CSSClassnames from '../utils/CSSClassnames';

const CLASS_ROOT = CSSClassnames.TAGS;

export default class Tags extends Component {

  render () {
    var classes = [CLASS_ROOT];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    var other = Props.pick(this.props, Object.keys(Box.propTypes));

    return (
      <Box {...other}
        className={classes.join(' ')}
        direction={this.props.direction}
        align={this.props.align}
        wrap={true}>
        {this.props.children}
      </Box>
    );
  }
}

Tags.propTypes = {
  ...Box.propTypes
};

Tags.defaultProps = {
  direction: 'row',
  align: 'start'
};
