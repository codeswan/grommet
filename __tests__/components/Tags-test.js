// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Tags from '../../src/js/components/Tags';
import Tag from '../../src/js/components/Tag';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Tags', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Tags className="custom-class">
        <Tag label="First" />
        <Tag label="Second" />
        <Tag label="Third" />
      </Tags>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct className rendering', () => {
    const component = renderer.create(
      <Tags className="testing">
        <Tag label="First" />
        <Tag label="Second" />
        <Tag label="Third" />
      </Tags>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
