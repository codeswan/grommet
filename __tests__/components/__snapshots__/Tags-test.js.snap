exports[`Tags has correct className rendering 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-tags testing"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-tag"
    onClick={undefined}>
    <a
      aria-label={undefined}
      className="grommetux-anchor grommetux-tag--label"
      href={undefined}
      id={undefined}
      onClick={undefined}
      target={undefined}>
      <span>
        First
      </span>
    </a>
  </div>
  <div
    className="grommetux-tag"
    onClick={undefined}>
    <a
      aria-label={undefined}
      className="grommetux-anchor grommetux-tag--label"
      href={undefined}
      id={undefined}
      onClick={undefined}
      target={undefined}>
      <span>
        Second
      </span>
    </a>
  </div>
  <div
    className="grommetux-tag"
    onClick={undefined}>
    <a
      aria-label={undefined}
      className="grommetux-anchor grommetux-tag--label"
      href={undefined}
      id={undefined}
      onClick={undefined}
      target={undefined}>
      <span>
        Third
      </span>
    </a>
  </div>
</div>
`;

exports[`Tags has correct default options 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-tags custom-class"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-tag"
    onClick={undefined}>
    <a
      aria-label={undefined}
      className="grommetux-anchor grommetux-tag--label"
      href={undefined}
      id={undefined}
      onClick={undefined}
      target={undefined}>
      <span>
        First
      </span>
    </a>
  </div>
  <div
    className="grommetux-tag"
    onClick={undefined}>
    <a
      aria-label={undefined}
      className="grommetux-anchor grommetux-tag--label"
      href={undefined}
      id={undefined}
      onClick={undefined}
      target={undefined}>
      <span>
        Second
      </span>
    </a>
  </div>
  <div
    className="grommetux-tag"
    onClick={undefined}>
    <a
      aria-label={undefined}
      className="grommetux-anchor grommetux-tag--label"
      href={undefined}
      id={undefined}
      onClick={undefined}
      target={undefined}>
      <span>
        Third
      </span>
    </a>
  </div>
</div>
`;
